assets/src/components/propertiesPanel/minischedule.tsx

```diff
@@ -189,31 +188,63 @@ const Piece = ({
             rightText={formattedScheduledTime(piece.start.time)}
           />
         ) : null}
-        {piece.trips.map((trip, index) => {
+        {piece.trips.map((trip, tripIndex) => {
           const sequence: "first" | "middle" | "last" =
-            index === 0
+            tripIndex === 0
               ? "first"
-              : index === piece.trips.length - 1
+              : tripIndex === piece.trips.length - 1
               ? "last"
               : "middle"
+          const tripTimeBasedStyle =
+            pieceTimeBasedStyle === "current"
+              ? getTimeBasedStyle(tripIndex, activeIndex && activeIndex[1])
+              : pieceTimeBasedStyle
+          const layoverTimeBasedStyle =
+            tripTimeBasedStyle === "current"
+              ? vehicleOrGhost.routeStatus === "on_route"
+                ? "past"
+                : "current"
+              : tripTimeBasedStyle
+          const onRouteTimeBasedStyle =
+            tripTimeBasedStyle === "current"
+              ? vehicleOrGhost.routeStatus === "on_route"
+                ? "current"
+                : "future"
+              : tripTimeBasedStyle
+
           return (
             <React.Fragment key={trip.startTime}>
+              {view === "run" ? (
+                <Layover
+                  nextTrip={trip}
+                  previousTrip={piece.trips[tripIndex - 1]}
+                  extraClasses={[
+                    `m-minischedule__row--${layoverTimeBasedStyle}`,
+                    layoverTimeBasedStyle === "current"
+                      ? drawnStatus(vehicleOrGhost)
+                      : null,
+                  ]}
+                />
+              ) : null}
               {isTrip(trip) ? (
                 <Trip
                   trip={trip}
                   sequence={sequence}
-                  vehicleOrGhost={vehicleOrGhost}
+                  extraClasses={[
+                    `m-minischedule__row--${onRouteTimeBasedStyle}`,
+                    onRouteTimeBasedStyle === "current"
+                      ? drawnStatus(vehicleOrGhost)
+                      : null,
+                  ]}
                 />
               ) : (
-                <AsDirected asDirected={trip} />
-              )}
-              {view === "run" ? (
-                <Layover
-                  currentTrip={trip}
-                  nextTrip={piece.trips[index + 1]}
-                  vehicleOrGhost={vehicleOrGhost}
+                <AsDirected
+                  asDirected={trip}
+                  extraClasses={[
+                    `m-minischedule__row--${onRouteTimeBasedStyle}`,
+                  ]}
                 />
-              ) : null}
+              )}
```

Comment:
This function is getting really big, but to pull out the function in the map would require a function with like 7 arguments. Any idea for a good way to break it into smaller pieces?

Reply:
Maybe a function with 7 arguments is the lesser evil here, this really is getting big.

Reply:
Got it down to 6 arguments. It's not perfect, but it is a little easier to see the big picture now.